manifests/init.pp

```diff
@@ -425,7 +418,7 @@
   }
   if $puppetrun_provider {
     validate_string($puppetrun_provider)
-    if $puppetrun_provider == 'puppetssh' and $puppet_split_config_files {
+    if $puppetrun_provider == 'puppetssh' {
       $real_puppetrun_provider = 'ssh'
       warning('foreman_proxy::puppetrun_provider should be "ssh", not "puppetssh" for 1.12 and above')
```

Comment:
I wanted, but was unsure about the timing... So it would be OK to change the warning to a fail and basically reverting most of 49fd8a39?